LiteCore/tests/LogEncoderTest.cc

```diff
@@ -288,6 +289,7 @@ TEST_CASE("Logging throw in c4", "[Log]") {
     sprintf(folderName, "Log_Rollover_%" PRIms "/", now.count());
     FilePath tmpLogDir = TestFixture::sTempDir[folderName];
     // Note that we haven't created tmpLogDir.
+    ExpectingExceptions x;
```

Comment:
Why do you use this? I tried to make c4log_writeToBinaryFile() not susceptible to the exception, so that the client code does not have to do something like above line.

Reply:
The Xcode project has a C++ exception breakpoint, so it will pause at any `throw`. Whether or not it's caught doesn't matter; this is at the instant before the stack unwinds. This is a pain when running unit tests that happen to throw an exception.

To work around that, there's a global boolean that the breakpoint tests; if the flag is set the breakpoint is skipped. `ExpectingExceptions` is a tiny class that just sets that flag while it's in scope.

Reply:
Is it this, 
struct ExpectingExceptions {
        ExpectingExceptions() {
            ++gC4ExpectExceptions;
            c4log_warnOnErrors(false);
        }
I had this in Logging.cc,
#define ERROR_THROW(ERROR_CODE, FMT, ...)                      \
    do {ScopedSetter setter(error::sWarnOnError, false);       \
        error::_throw(ERROR_CODE, FMT, ## __VA_ARGS__);} while (false)
It does the same thing as ExpectingExceptions, but doing it where the exception is thrown. If this line is needed here, I am afraid somewhere else there is thrown.